src/wp-includes/class-wp-block.php

```diff
@@ -492,13 +492,12 @@ public function replace_rich_text( $rich_text ) {
 	public function render( $options = array() ) {
 		global $post;
 
-		// Capture the current assets queues and then clear out to capture the diff of what was introduced by rendering.
+		$before_wp_head_count = did_action( 'wp_head' );
```

Comment:
In @dd32's code, it appears that the specific problem is not `wp_head` but rather `wp_enqueue|register_style|scripts`. It just happens that `wp_head` fires them.

It could be best to test for the specific rather than the general. 🤷‍♂️

Reply:
The issue (as I found) is with `wp_print_scripts()` and `wp_print_styles()`, that either of these (especially the former) may not print anything during `wp_head`, since what is enqueued may get printed in `wp_footer`. We could change this to look if the `wp_print_scripts` or `wp_print_styles` actions fired, but they don't always fire.

Or actually, maybe it would be better to check if the `wp_enqueue_scripts` action fired. As I think about it, the issue is that plugins and themes are being invited to enqueued scripts and styles, but if nothing is printed then we undo the enqueueing. So this needs to be prevented.